client/reducers/currentOrder.js

```diff
@@ -0,0 +1,49 @@
+import axios from 'axios';
+
+// ACTION TYPES 
+const SET_ORDER = 'SET_ORDER';
+const ADD_ITEM = 'ADD_ITEM';
+const DELETE_ITEM = 'DELETE_ITEM';
+const UPDATE_ITEM = 'UPDATE_ITEM';
+
+// ACTION CREATORS
+const setOrder = order => {
+  return {type: SET_ORDER, order}
+}
+
+const addNewItem = item => {
+  return {type: ADD_ITEM, item}
+}
+
+const deleteItem = item => {
+  return {type:DELETE_ITEM, itemId}
+}
+
+const updateExistingItem = item => {
+  return {type: UPDATE_ITEM, item}
+}
+
+
+// REDUCER
+export default (currentOrder = [], action) => {
+  switch(action.type) {
+    case SET_ORDER: 
+      return action.order; 
+    case ADD_ITEM: 
+      return [action.item, ...currentOrder];
+    case DELETE_ITEM:
+      return currentOrder.filter(item => item.id !== action.itemId);
+    case UPDATE_ITEM:
+      return currentOrder.map(item => (item.id === action.item.id ? action.item : item));
+    default: return currentOrder; 
+  }
+}
+
+//THUNK 
+export const setCurrentOrder = order => dispatch => {
```

Comment:
do we need to change how the info coming back from AJAX request looks like so that the product quantity is correct for how we want to represent it on front end store